api/v1alpha4/vspherecluster_webhook.go

```diff
@@ -28,17 +29,23 @@ func (c *VSphereCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha4-vspherecluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=vsphereclusters,versions=v1alpha4,name=validation.vspherecluster.infrastructure.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+// +kubebuilder:webhook:verbs=create;update,path=/default-infrastructure-cluster-x-k8s-io-v1alpha4-vspherecluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=vsphereclusters,versions=v1alpha4,name=default.vspherecluster.infrastructure.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha4-vspherecluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=vsphereclusters,versions=v1alpha4,name=validation.vspherecluster.infrastructure.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
+
+var _ webhook.Defaulter = &VSphereCluster{}
+
+func (c *VSphereCluster) Default() {
+	defaultVSphereCluterSpec(&c.Spec)
+}
```

Comment:
Is there a specific reason that the defaulting webhook is being included here? I see the implementation of the method is empty. If this is for the sake of future usage, let's introduce it only when there is some specific defaulting logic that needs to be implemented.

Reply:
Yes, it is for future use cases. So that we make sure to introduce consistent defaulting for template and the cluster type when ever we do that.

Reply:
My vote is to wait till the need arises for a separate webhook. Adding a note in the clustertemplate_webhook to make sure this needs to be replicated across the two types would be a way to move forward. 
Thoughts @gab-satchi @yastij ?

Reply:
+1 on not adding dead code

Reply:
Thanks for the feedback @srm09 @yastij. Will update the PR :)